hibernate-core/src/main/java/org/hibernate/collection/spi/AbstractPersistentCollection.java

```diff
@@ -652,6 +656,12 @@ private void throwLazyInitializationException(String message) {
 		);
 	}
 
+	public static void checkPersister(PersistentCollection collection, CollectionPersister persister) {
+		if ( !collection.wasInitialized() && persister == null ) {
+			((AbstractPersistentCollection) collection).throwLazyInitializationException( "collection is being removed" );
+		}
+	}
```

Comment:
Does this method really need to be `static`?

Reply:
Do you want to add a default method to the PersistenCollection interface? Even if it's marked as @Internal, it's still poorly justified i think.

Reply:
No I was asking if it can be an instance method so that you don't need to pass in the `PersistentCollection collection` and then cast it to `AbstractPersistentCollection`.

Reply:
That's what i mean, if we don't add it to the super interface, we end up having to cast in `DefaultInitializeCollectionEventListener`, no sure which is better, but now at least it's contained so to speak?